### PR TITLE
feat: updated the secure banner

### LINKED ISSUE #28549: [Feature]: Update banner on create new datasource page
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

https://www.notion.so/appsmith/Task-breakdown-Start-with-data-787dd729e71e452abe14b46acae6e931?p=8879f1a2c92e41aa89dc59da88a7a4e4&pm=s

### Why should this be worked on?

activation - onboarding revamp

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description
Updated the design of the secure banner that comes on the datasources list page. Allowed it to be shown on the onboarding start with data flow as well.

Previously :
Screenshot 2023-11-21 at 4 23 43 PM

Now :
Screenshot 2023-11-21 at 4 23 25 PM

#### PR fixes following issue(s)
Fixes #28549 
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/pages/Editor/IntegrationEditor/AddDatasourceSecurely.tsx [hunk 1]
```diff
@@ -1,60 +1,32 @@
...
-        </Content>
-      </div>
-      <img className="datasource-img" src={AppsmithDatasource} />
+      <img
+        alt="Secure & fast connection"
+        src={getAssetUrl(`${ASSETS_CDN_URL}/secure-lock.svg`)}
+      />
+      <Flex flexDirection="column" ml="spaces-4">
+        <Text color="var(--ads-v2-color-gray-700)" kind="heading-m">
+          Secure & fast connection
```
Comment (sneha122) [inline app/client/src/pages/Editor/IntegrationEditor/AddDatasourceSecurely.tsx:23]: Can we please avoid magic strings? We can store these in messages.ts

Comment (AmanAgarwal041) [reply app/client/src/pages/Editor/IntegrationEditor/AddDatasourceSecurely.tsx]: Updated the title in messages.ts but the description has an underline text which cannot be implemented in message.ts hence added it as magic string

**Review by sneha122 — APPROVED** [review]

**Comment by sneha122** [comment]
@AmanAgarwal041 I have tested the DP, looks good, just noticed one thing: Font width of heading looks a bit different than designs can you please check that? [Designs](https://www.figma.com/file/kbU9xPv44neCfv1FFo9Ndu/User-Activation?type=design&node-id=2365-20296&mode=design&t=p9OBkXUQDOxToWji-0)

Screenshot 2023-11-27 at 2 03 20 PM


Screenshot 2023-11-27 at 2 04 49 PM

**Comment by AmanAgarwal041** [comment]
Checked with @carinanfonseca, the designs look good. However the designs for the section title on datasource list screen uses old design system components. Will be opening new issue to tackle this. Hence, adding test plan approved label.

**Pull request merged**